### DISCUSSION TITLE
Javascript Debugging Tools

### DISCUSSION THREAD

**Discussion by TheFunkyBiskit** [Budibase/budibase#9332] [Ideas]
I write a lot of Javascript snippets for various things within Budibase, and it is fantastic to have that flexibility. However one thing I find to be very cumbersome is the lack of any kind of debug tools or a way to log out information. For example, just being able to see the output of the $() functions for retrieving data from other components. I understand that these code snippets are executed server-side, but is there any way that the output can be piped through to the builder so that console.log() outputs appear in the browser's console? My current solution is to use a text block and set the text to the script I'm interest in, then return any value I want to inspect. But this is obviously not optimal, and has no way of displaying the structure of an object, which is often what I'm working with in these scripts.

**Comment by melohagan** [comment]
Completely agree! It can be a little frustrating/unforgiving sometimes.

There's a couple of related issues here:
 - https://github.com/Budibase/budibase/issues/3816
 - https://github.com/Budibase/budibase/issues/4908

There is also some mention of this in the 2023 roadmap for Q2: https://github.com/Budibase/budibase/projects/26

**Comment by jjchmielowiec** [comment]
I agree this is a big omission.

**Comment by Saladien** [comment]
Agree, im actually shocked that this is not in because a professional tool just needs such features.
